src/setup-java.ts

```diff
@@ -27,6 +29,7 @@ async function run() {
       throw new Error(`No supported distribution was found for input ${distributionName}`);
     }
 
+    const restoreResult = cache ? restore(cache) : Promise.resolve();
```

Comment:
Could you please make it via if check and move it after configureAuthentication. As I know promise.all keep order for returning of the array but not for execution. I would prefer to configure authentication firstly.

Reply:
Let me confirm: is the following change satisfy your request?

```suggestion
    await auth.configureAuthentication();
    await restoreResult;
```

Or do we need to move `restore(cache)` to after `configureAuthentication`? This change probably makes the benefit of this PR less, because we cannot run JDK installation and restoring the dependency cache in parallel.

Reply:
Confirmed that performance isn't good if [we run the restore process after the auth ends](https://github.com/KengoTODA/setup-java/commit/bd074812489119ed6274bc5c17b836e16f3a612a).
Its 95 percentile is [5% slower than `actions/cache + actions/setup-java`](https://docs.google.com/spreadsheets/d/1Q7OQJ7UuVNg78FeNhc1DfMAJGJnKJoCHTXQibfMmrlE/edit#gid=1914328659&range=D37).

So if we need to run the cache after the auth, I think this PR isn't so valuable for performance.

Reply:
I've rechecked the performance after we fix the cache process on windows, and confirmed that the performance isn't bad; [21% faster than `actions/cache + actions/setup-java`](https://docs.google.com/spreadsheets/d/1Q7OQJ7UuVNg78FeNhc1DfMAJGJnKJoCHTXQibfMmrlE/edit#gid=2134038139&range=D37).

![Perf  comparison for Java cache (1)](https://user-images.githubusercontent.com/505751/125712424-693834db-a61d-4256-906c-d3943e1411be.png)
I'll push 6977c031ae59b16fb87b34271efc3bd5fb24b057 to make the restore process running after the auth.